Add free-form text to qube for notes, comments, ...

alimirjamali opened the pull request:
qubesadmin part of adding free-form text to each qube for comments, notes, descriptions, remarks, reminders, etc.

fixes: https://github.com/QubesOS/qubes-issues/issues/899

alimirjamali committed:
Add free-form text to qube for notes, comments, ...

qubesadmin part of adding free-form text to each qube for comments,
notes, descriptions, remarks, reminders, etc.

fixes: https://github.com/QubesOS/qubes-issues/issues/899

Additional options to skip legacy backup tests or super slow tests

marmarek commented on qubesadmin/vm/__init__.py:
```diff
@@ -394,6 +394,23 @@ def klass(self):
+        ''' Get qube notes '''
+        try:
+            response = self.qubesd_call(self._method_dest, 'admin.vm.notes.Get')
+        except Exception as exc:
```
This is definitely too broad. If anything, maybe just `qubesadmin.exc.QubesDaemonNoResponseError`, but even that I'm not sure. qubesd should return already correct exception on error.

alimirjamali replied:
This is done

marmarek commented on qubesadmin/vm/__init__.py:
```diff
@@ -394,6 +394,23 @@ def klass(self):
+        try:
+            self.qubesd_call(self._method_dest, 'admin.vm.notes.Set', \
+                payload = str(notes).encode('ascii'))
+        except Exception as exc:
```
and same here

alimirjamali replied:
This is also done

alimirjamali commented:
@marmarek please let me know if you agree with `qvm-notes` syntax and options. So I can proceed with its unittests. in the meantime, I will write the unittests for the rest.

marmarek commented:
Generally qvm-notes interface looks fine (based on its man page). I think append action needs a clarification regarding newlines if any are inserted automatically.

As for the implementation, IMO too many of the actions ask for confirmation. Maybe remove should, but others seems unnecessary.

alimirjamali commented:
This is ready for review. Only 3 lines are not covered by tests but two lines are abstract lines (maybe I should add `#pragma: no cover`) and one other is for backup restore notes failure which is highly unlikely to happen (this is covered in lots of other places).

> Generally qvm-notes interface looks fine (based on its man page). I think append action needs a clarification regarding newlines if any are inserted automatically.

This is done. I also added some more notes for better clarification

> 
> As for the implementation, IMO too many of the actions ask for confirmation. Maybe remove should, but others seems unnecessary.

Theoretically It is possible to check with `get_notes` and only ask for confirmation on `set, delete & import` actions that the action is going to replace existing note. On the other hand, it will require `admin.vm.notes.Get` permission. So I implemented your recommendation.

alimirjamali commented:
@marmarek I just rebased this and excluded few irrelevant lines for codecov. Otherwise nothing is changed.
